### ISSUE TITLE
Clearing cache across environments

### ISSUE THREAD

**Issue by pushred** [SparkartGroupInc/qa-deployer#18]
Sometimes a build in CircleCI can run into issues due to some cached dependencies. We can rebuild with the cache cleaned but if we're not deploying to Modulus with the `--include-modules` flag then the same issues can still occur in that environment. We're currently not including modules in our deploys because this is still slow and prone to timeouts, even with the new build system.

Generally deploys are working more reliably with modules now as well, perhaps due to increased use of caret versioning, using private modules and git tags versus branches, etc. But I've still seen cache-related issues that take a couple of attempts to shake out (unless the issue is totally intermittent).

As a sanity check, it would be great if we could trigger a ["force push" that skips the cache][1] without having to change a project's configuration. Ideally CircleCI's own [Rebuild Without Cache][2] option could be the trigger. We could detect this in the build by examining the contents of `~/.npm`. If it contains directories other than one named `-` then cache is present and we should deploy normally. If it's clear then it's likely that this is the very first build or that the rebuild without cache option was used.

[1]: https://github.com/onmodulus/modulus-cli/blob/master/lib/routes/project.js#L156
[2]: https://circleci.com/docs/how-cache-works

**Comment by joanniclaborde** [comment]
A quick search across our sites (the ones I have cloned locally, at least) reveals that almost all non-versioned dependencies are `broadway` and `jungle-solidus`. And with the new private modules, these should be changed pretty soon, if it's not already done.

The remaining dependencies are git URLs, and as we experienced last week, the newer npm versions reinstall those for every `npm install`.

Given this, I suggest we wait a few days before considering this change, to see if we ever use the Rebuild Without Cache button again.
